### ISSUE TITLE
prefer-locator checks only for page keyword and method name

### ISSUE THREAD

**Issue by danio768** [playwright-community/eslint-plugin-playwright#344]
I have property that start with page and same name of built-in method.
Simple repro:
```
class PageClass {
    pageFooter: PageClass;
    footerPage: PageClass;
    footer: PageClass;

    constructor() {
        this.pageFooter = new PageClass();
        this.footerPage = new PageClass();
        this.footer = new PageClass();
    }

    isHidden() {}

    // Prefer locator methods instead of page methods
    is1() {
        this.pageFooter.isHidden();
    }

    // Prefer locator methods instead of page methods
    is2() {
        this.footerPage.isHidden();
    }

    // Is fine
    isFine() {
        this.footer.isHidden();
    }
}
```
Rule:
```'playwright/prefer-locator': 'error'```

**Comment by carlbray** [comment]
I'm not sure what you're trying to achieve?

page or frame is detected by isPageMethod(node: ESTree.CallExpression, name: string)
https://github.com/playwright-community/eslint-plugin-playwright/blob/main/src/utils/ast.ts#L119

**Comment by danio768** [comment]
I don't want to achieve anything just bump on it when I enabled this rule in my project and one of properties have page in name and found it weird that it was detected as problem. 
I just got one case so this no big problem for me but I thought that I will raise it here. 
You can close it if it is working as expected.

**Comment by mskelton** [comment]
This is expected behavior

**Comment by ochubey** [comment]
@mskelton let me try to provide a bit different example with JS.
This code will trigger a `playwright/prefer-locators` rule:
```js
export class SomePageObject {
    /**
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        /** @type {import('@playwright/test').Locator} */
        this.navigateToSomeOtherPage = this.page.locator('div.fake_class');
    }

    async toSomeOtherPage() {
        await this.navigateToSomeOtherPage.dblclick();
        return new SomeOtherPage(this.page);
    }
}
```

But if we rename `navigateToSomeOtherPage` to `navigateToSomeOtherPages` - rule would not be trigerred.
Please consider to add this as a knowing issue, since, as I understand, there is no robust way to get runtime type of the caller from AST node.
